core/components/tree/treenode.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2019 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Definition of the Blockly.tree.TreeNode class.
+ * This class is similar to Closure's goog.ui.tree.TreeNode class.
+ * @author samelh@google.com (Sam El-Husseini)
+ */
+'use strict';
+
+goog.provide('Blockly.tree.TreeNode');
+
+goog.require('Blockly.tree.BaseNode');
+goog.require('Blockly.utils.KeyCodes');
+
+
+/**
+ * A single node in the tree, customized for Blockly's UI.
+ * Similar to Closure's goog.ui.tree.TreeNode
+ *
+ * @param {Blockly.Toolbox} toolbox The parent toolbox for this tree.
+ * @param {string} content The content of the node label treated as
+ *     plain-text and will be HTML escaped.
+ * @param {Object} config The configuration for the tree.
+ * @constructor
+ */
+Blockly.tree.TreeNode = function(toolbox, content, config) {
+  this.toolbox_ = toolbox;
+  Blockly.tree.BaseNode.call(this, content, config);
+};
+goog.inherits(Blockly.tree.TreeNode, Blockly.tree.BaseNode);
+
+/**
+ * Returns the tree.
+ * @return {?Blockly.Toolbox.TreeControl} The tree.
+ * @override
+ */
+Blockly.tree.TreeNode.prototype.getTree = function() {
+  if (this.tree) {
+    return this.tree;
+  }
+  var parent = this.getParent();
+  if (parent) {
+    var tree = parent.getTree();
+    if (tree) {
+      this.setTreeInternal(tree);
+      return tree;
+    }
+  }
+  return null;
+};
+
+/**
+ * Returns the source for the icon.
+ * @return {string} Src for the icon.
+ * @override
+ */
+Blockly.tree.TreeNode.prototype.getCalculatedIconClass = function() {
```

Comment:
Is this the same as the one for the tree control?

Reply:
Nope, different CSS used in each case.